server/src/main/java/io/crate/execution/dml/RawIndexer.java

```diff
@@ -55,6 +56,9 @@ public class RawIndexer {
 
     private final Map<Set<String>, Indexer> indexers = new HashMap<>();
 
+    private Indexer currentRowIndexer;
+    private Object[] currentRowValues;
+
     public RawIndexer(String indexName,
```

Comment:
To avoid record.

In "normal" indexer, we need only newColumns, otherwise we already have values/indexer object.

Here we need a "complex" state after the phase 1 - "normal" indexer, either cached or created, values parsed from doc. 

After the discussion I thought it would be easier to introduce 2 instances which phase 2 will be utilizing in a dedicated index() method.